packages/syft/src/syft/client/client.py

```diff
@@ -1060,3 +1035,17 @@ def get_client(
     @classmethod
     def get_client_for_node_uid(cls, node_uid: UID) -> Optional[SyftClient]:
         return cls.__client_cache__.get(node_uid, None)
+
+
+T = TypeVar("T")
+P = ParamSpec("P")
+
+
+def _prompt_for_password(f: Callable[P, T]) -> Callable[P, T]:
+    def inner(*args: P.args, **kwargs: P.kwargs):
+        if kwargs.get("password") is None:
+            kwargs["password"] = getpass("Password: ")
+
+        return f(*args, **kwargs)
+
+    return inner
```

Comment:
One small issue with is now if the user wants to pass password directly as an argument to `client.images.push` they have to pass it as a kwarg. Passing it as a positional arg would not work, but this should be fixable (by getting the signatures of `f` and check which argument is `password`).